spec/wibox/layout/align_spec.lua

```diff
@@ -223,6 +223,190 @@ describe("wibox.layout.align", function()
+        end)
+
+        it("empty layout layout", function()
+            assert.widget_layout(layout, { 0, 0 }, {})
+        end)
+
+        describe("with widgets", function()
+            describe("with equal sides", function()
+                local first, second, third
+
+                before_each(function()
+                    first = utils.widget_stub(10, 10)
+                    second = utils.widget_stub(15, 15)
+                    third = utils.widget_stub(10, 10)
+
+                    layout:set_first(first)
+                    layout:set_second(second)
+                    layout:set_third(third)
+                end)
+
+                describe("with enough space", function()
+                    it("fit", function()
+                        assert.widget_fit(layout, { 100, 100 }, { 15, 35 })
+                    end)
+
+                    it("layout", function()
+                        assert.widget_layout(layout, { 100, 100 }, {
+                            p(first,  0,  0, 100, 10),
+                            p(third,  0, 90, 100, 10),
+                            p(second, 0, 10, 100, 80),
+                        })
+                    end)
+                end)
+
+                describe("without enough height", function()
+                    it("fit", function()
+                        assert.widget_fit(layout, { 5, 100 }, { 5, 35 })
+                    end)
+
+                    it("layout", function()
+                        assert.widget_layout(layout, { 5, 100 }, {
+                            p(first,  0,  0, 5, 10),
+                            p(third,  0, 90, 5, 10),
+                            p(second, 0, 10, 5, 80),
+                        })
+                    end)
+                end)
+
+                describe("without enough width", function()
+                    it("fit", function()
+                        assert.widget_fit(layout, { 100, 20 }, { 15, 20 })
+                    end)
+
+                    it("layout", function()
+                        assert.widget_layout(layout, { 100, 20 }, {
+                            p(first,  0,  0, 100, 10),
+                            p(third,  0, 10, 100, 10),
+                        })
+                    end)
+                end)
+            end)
+
+            describe("with first larger", function()
+                local first, second, third
+
+                before_each(function()
+                    first = utils.widget_stub(10, 20)
+                    second = utils.widget_stub(15, 15)
+                    third = utils.widget_stub(10, 10)
+
+                    layout:set_first(first)
+                    layout:set_second(second)
+                    layout:set_third(third)
+                end)
+
+                describe("with enough space", function()
+                    it("fit", function()
+                        assert.widget_fit(layout, { 100, 100 }, { 15, 45 })
```

Comment:
Yes, it should be `55`. Since this tests was passing with `45`, I imagine there might be an error in the code as well. I will fix it as soon as I get a chance.

Reply:
Yes it should have been `55`, I didn't notice that I also needed to remake the `align:fit` function, because it should report different size for the `"justify"` mode. I fixed it in [9ac156d](https://github.com/awesomeWM/awesome/pull/3588/commits/9ac156db6d3c5eef7a3c02f9cff7a4a1a01f33d2), and will squash it with previous commits so it doesn't have the bug from the start.